### ISSUE TITLE
Looping Functions, Understanding "Top," & Using BuildInnerBreakdownsRecursively 

### ISSUE THREAD

**Issue by cschweitzer** [randyzwitch/RSiteCatalyst#145]
Hello!
I have a few questions that I'm hoping are related:

I am using the QueueTrended function to get a report from SiteCatalyst, but I am limited by the API row limit. To overcome this, I set top as 50,000 and created a loop function that increments the “start” value by the number of rows previously retrieved and appends it to the data frame. This worked great when date.granularity was greater than the date range (basically producing an aggregate report).
However, when I query a smaller granularity, it breaks. First, I get more rows than the “top” when I run the first request. I lowered my top value from 50,000 to 10,000 and the output has 90,000 rows (top=5000 > output = 25000). I think it has to do with this: https://github.com/randyzwitch/RSiteCatalyst/blob/master/R/QueueRanked.R#L94-10
Is that correct?

I also get the following error after it has completed successful iterations of the loop:
Error in names(counts.df) <- metrics : 'names' attribute [3] must be the same length as the vector [0]

I see others have the same error, though mine is attribute 3. What do the attribute numbers refer to?
I’ve been trying to wrap my head around the BuildInnerBreakdownsRecursively discussion, but I don’t understand how to use the function. Do you have an example? Is it meant to solve the problem of my loop above? Or would I use it in combination with my loop to fix the error that I’m getting.

This is the basic outline of my QueueTrended report, with variations around start and top:

fdc_users_weekly <-  
  QueueTrended(reportingsuite,"2015-07-01","2015-07-31",
               metrics=c("visits", "uniquevisitors", "pageviews"),
               date.granularity = "week",
               elements = c("evar10"),
               search = ("::unspecified::"),search.type = "NOT"
               #,start=0
               ,top = 50000
               )

Thank you in advance!
Katie

**Comment by randyzwitch** [comment]
>I am using the QueueTrended function to get a report from SiteCatalyst, but I am limited by the API row limit. To overcome this, I set top as 50,000 and created a loop function that increments the “start” value by the number of rows previously retrieved and appends it to the data frame. This worked great when date.granularity was greater than the date range (basically producing an aggregate report).
However, when I query a smaller granularity, it breaks. First, I get more rows than the “top” when I run the first request. I lowered my top value from 50,000 to 10,000 and the output has 90,000 rows (top=5000 > output = 25000). I think it has to do with this: https://github.com/randyzwitch/RSiteCatalyst/blob/master/R/QueueRanked.R#L94-10
Is that correct?

This isn't "broken" per se; top refers to the number of results of the first element. When specifying more than one element, you currently get the "top" number of elements of the first element and up to 50,000 of the other elements (subrelations or correlations in Adobe Analytics speak).

People have asked to control how many results come back from second and third levels of the `elements` parameter. All it would take for this to happen would be for someone to write the code and submit a pull request.

>I also get the following error after it has completed successful iterations of the loop:
Error in names(counts.df) <- metrics : 'names' attribute [3] must be the same length as the vector [0]

>I see others have the same error, though mine is attribute 3. What do the attribute numbers refer to?

The number refers to the number of columns of counts.df; the error is trying to overwrite the names of a null dataframe, which is the R error you see.

In brief, the error is because the results you request from Adobe are such deeply nested JSON and eventually there's a null level. @WillemPaling has been trying to fix this, but at some point, I'm inclined to blame Adobe for having JSON that is so hard to parse

>I’ve been trying to wrap my head around the BuildInnerBreakdownsRecursively discussion, but I don’t understand how to use the function. Do you have an example? Is it meant to solve the problem of my loop above? Or would I use it in combination with my loop to fix the error that I’m getting.

`BuildInnerBreakdownsRecursively` is an internal function, not one meant to solve any publicly-facing user issues.

**Comment by cschweitzer** [comment]
Thanks @randyzwitch, I appreciate you getting back to me. That all makes sense!

Best,
Katie
